test/WebApi20.FunctionalTests/FunctionalTest/MultipleWebHostsTests.cs

```diff
@@ -0,0 +1,140 @@
+
+                var request = telemetry.Single(t => t is TelemetryItem<RequestData>);
+                Assert.Equal("200", ((TelemetryItem<RequestData>)request).data.baseData.responseCode);
+            }
+
+            using (var server2 = new InProcessServer(assemblyName, this.output))
+            {
+                this.ExecuteRequest(server2.BaseHost + requestPath);
+                var telemetry = server2.Listener.ReceiveItems(TestListenerTimeoutInMs);
+                this.DebugTelemetryItems(telemetry);
+
+                Assert.Single(telemetry.Where(t => t is TelemetryItem<RequestData>));
+                Assert.Single(telemetry.Where(IsServiceDependencyCall));
+                Assert.DoesNotContain(telemetry, t => t is TelemetryItem<ExceptionData>);
+
+                var request = telemetry.Single(t => t is TelemetryItem<RequestData>);
+                Assert.Equal("200", ((TelemetryItem<RequestData>)request).data.baseData.responseCode);
+            }
+        }
+
+        [Fact(Skip = "We track each request and depednency by each WebHost, issue #621")]
+        public void TwoWebHostsCreatedInParallel()
+        {
+            using (var server1 = new InProcessServer(assemblyName, this.output))
+            using (var server2 = new InProcessServer(assemblyName, this.output))
+            {
+                this.ExecuteRequest(server1.BaseHost + requestPath);
+                var telemetry1 = server1.Listener.ReceiveItems(TestListenerTimeoutInMs);
+                this.DebugTelemetryItems(telemetry1);
+
+                this.ExecuteRequest(server2.BaseHost + requestPath);
+                var telemetry2 = server2.Listener.ReceiveItems(TestListenerTimeoutInMs);
+
+                this.DebugTelemetryItems(telemetry2);
+                Assert.Single(telemetry1.Where(t => t is TelemetryItem<RequestData>));
+                Assert.Single(telemetry1.Where(IsServiceDependencyCall));
+                Assert.DoesNotContain(telemetry1, t => t is TelemetryItem<ExceptionData>);
+
+                var request1 = telemetry1.Single(t => t is TelemetryItem<RequestData>);
+                Assert.Equal("200", ((TelemetryItem<RequestData>)request1).data.baseData.responseCode);
+
+                // Fails here, we track everything twice
+                // it did not happen with the first host because second one has not been really started yet
+                Assert.Single(telemetry2.Where(t => t is TelemetryItem<RequestData>));
+                Assert.Single(telemetry2.Where(IsServiceDependencyCall));
+                Assert.DoesNotContain(telemetry2, t => t is TelemetryItem<ExceptionData>);
+
+                var request2 = telemetry2.Single(t => t is TelemetryItem<RequestData>);
+                Assert.Equal("200", ((TelemetryItem<RequestData>)request2).data.baseData.responseCode);
+            }
+        }
+
+        [Fact]
+        public void TwoWebHostsOneIsDisposed()
```

Comment:
Do we need a case for both being disposed? `Active` may still stay with its default channel on, will it hurt anything?

Reply:
Active will stay, as Active lifetime is a process lifetime. It's a good idea to test it still works after all web hosts are gone as in ServiceFabric SharedProcess mode, some background tasks are not bound to any web host and run before and after web hosts are created.